Chapters/Common/StringAlgorithms.cs

```diff
@@ -0,0 +1,315 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Chapters.Common
+{
+    public class StringAlgorithms
+    {
+
+        private string Input { get; set; }
+
+        public StringAlgorithms(string input)
+        {
+            Input = input;
+        }
+
+       
+        //private StringAlgorithms(Stream input)
+        //{
+        //    Input = new char[input.Length];
+        //    input.Read(Input, 0, Input.Length);
+        //}
+
+        public static StringAlgorithms Create(Chapter9TaskType type, object param = null)
+        {
+            switch (type)
+            {
+                case Chapter9TaskType.FromString:
+                    return new StringAlgorithms(param as string);
+
+                //case Chapter9TaskType.FromFile:
+                //    return new StringAlgorithms(param as Stream);
+
+                default:
+                    throw new ArgumentOutOfRangeException(type.GetType().Name, type, null);
+            }
+
+        }
+
+        //1
+        public int PunctuationCount()
+        {
+            int sum = 0;
+            foreach (var s in Input)
+                if(char.IsPunctuation(s))
+                        sum++;
+            return sum;
+        }
+
+        //2
+        public int PointExclamationPointCount()
+        {
+            int sum = 0;
+            foreach (var c in Input)
+                if (c == '.' || c == '!')
+                    sum++;
+            return sum;
+        }
+        
+        //3
+        public void RemoveAllCommas()
+        {
+            for (var i = 0; i < Input.Length; i++)
+                if (Input[i] == ',')
+                    Input.Remove(i, 1);
+        }
+        
+        //4
+        public void RemoveSpacesBeforePunctuation()
+        {
+            for (int i = 1; i < Input.Length; i++)
+                if (Input[i] == ' ' && char.IsPunctuation(Input[i - 1]))
+                    Input.Remove(i, 1);
+        }
+        
+        //5
+        public void InsertNumberAfterSpaces(int num)
+        {
+            for (int i = 0; i < Input.Length - 1; i++)
+                if (Input[i] == ' ')
+                    Input.Insert(i + 1, num.ToString());
```

Comment:
This is not a bug, but try to optimize this. There should be redundant iterations, and this is a bicycle.